src/pretix/plugins/stripe/views.py

```diff
@@ -467,3 +474,37 @@ def _redirect_to_order(self):
             'order': self.order.code,
             'secret': self.order.secret
         }) + ('?paid=yes' if self.order.status == Order.STATUS_PAID else ''))
+
+
+@method_decorator(xframe_options_exempt, 'dispatch')
+class ScaView(StripeOrderView, View):
+    def get(self, request, *args, **kwargs):
+        prov = self.pprov
+        prov._init_api()
+
+        payment_info = json.loads(self.payment.info)
+
+        if 'id' in payment_info:
+            intent = stripe.PaymentIntent.retrieve(
+                payment_info['id'],
+                **prov.api_kwargs
+            )
+        else:
+            messages.error(self.request, _('Sorry, there was an error in the payment process. Please check the link '
+                                           'in your emails to continue.'))
+            return redirect(eventreverse(self.request.event, 'presale:event.index'))
+
+        if intent.status == 'requires_action' and intent.next_action.type == 'use_stripe_sdk':
+            r = render(request, 'pretixplugins/stripe/sca.html', {
+                'stripe_settings': StripeSettingsHolder(self.order.event).settings,
+                'payment_intent_client_secret': intent.client_secret,
+            })
+            r._csp_ignore = True
+            return r
+        else:
+            StripeCC._handle_payment_intent(prov, request, self.payment)
```

Comment:
First we retrieve the payment intent, then we call ``_handle_payment_intent``, which retrieves it from the API again. This feels wasteful, any ideas to optimize it?